### PR TITLE
Change literal status flags to enum.

### PR DESCRIPTION
**Pull request by zherczeg**
Also fix a typo.

### CODE REVIEW

File: jerry-core/parser/js/js-parser-limits.h [hunk 1]
```diff
@@ -77,7 +77,7 @@
 #endif /* !PARSER_MAXIMUM_CODE_SIZE */
 
 /**
- * Maximum number of values pushed onto the stack by a function.
+ * Maximum number of values pushed onto the stack by a function.
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-limits.h]: There is a 'C2 A0' character here, which a "NO-BREAK SPACE'. Change it to a normal space.

File: jerry-core/parser/js/common.h [hunk 2]
```diff
@@ -55,24 +55,22 @@ typedef enum
...
-/** No space is allocated for this character literal. */
-#define LEXER_FLAG_SOURCE_PTR 0x40
-/** Initialize this variable after the byte code is freed. */
-#define LEXER_FLAG_LATE_INIT 0x80
+/**
+ * Flag bits for status_flags member of lexer_literal_t.
+ */
+typedef enum
+{
+  LEXER_FLAG_VAR = (1 << 0), /** local identifier (var, function arg) */
```
Comment (akosthekiss) [inline jerry-core/parser/js/common.h]: All the enum value comments need to be `/**<` as they are after the entity they are documenting.

**Comment by zherczeg** [comment]
Thank you for the review. Patch updated.

File: jerry-core/parser/js/common.h [hunk 3]
```diff
@@ -55,24 +55,22 @@ typedef enum
...
+  LEXER_FLAG_NO_REG_STORE = (1 << 1), /**< this local identifier cannot be stored in register */
+  LEXER_FLAG_INITIALIZED = (1 << 2), /**< this local identifier is initialized with a value */
+  LEXER_FLAG_FUNCTION_NAME = (1 << 3), /**< this local identifier has a reference to the function itself */
+  LEXER_FLAG_FUNCTION_ARGUMENT = (1 << 4), /**< this local identifier is a function argument */
+  LEXER_FLAG_UNUSED_IDENT = (1 << 5), /**< this identifier is referenced by sub-functions,
+                                       *   but not referenced by the currently parsed function */
+  LEXER_FLAG_SOURCE_PTR = (1 << 6), /**< the literal is directly referenced in the source code
+                                     *   (no need to allocate memory) */
+  LEXER_FLAG_LATE_INIT = (1 << 7), /**< initialize this variable after the byte code is freed */
+} parser_lexer_literal_status_flags_t;
```
Comment (akosthekiss) [inline jerry-core/parser/js/common.h]: Strange name for this type. It is used in `lexer_literal_t`, all the enum values have a `LEXER_` prefix, yet this type has a `parser_` prefix. I'd suggest going for `lexer_literal_status_flags_t` only.

Comment (zherczeg) [reply jerry-core/parser/js/common.h]: Good catch. I haven't noticed that.

**Comment by AnthonyCalandra** [comment]
LGTM (informal)

**Pull request merged**